for_each_test.go

```diff
@@ -55,3 +56,39 @@ func TestForEachContext(t *testing.T) {
 		require.Equal(t, testCase.Out, stringSlice)
 	}
 }
+
+func TestForEachContextError(t *testing.T) {
+	type contextKey string
+	var key = contextKey("key")
+	input := []string{"aaa", "bbb"}
+
+	var calledInputs = make([]string, 0, len(input))
+	fn := func(c context.Context, s string) error {
+		require.Equal(t, "a_value", c.Value(key))
+		calledInputs = append(calledInputs, s)
+		return nil
+	}
+	ctx := context.WithValue(context.Background(), key, "a_value")
+
+	err := ForEachContextError(ctx, input, fn)
+	require.Nil(t, err)
+	require.Equal(t, input, calledInputs)
+}
+
+func TestForEachContextErrorWithError(t *testing.T) {
```

Comment:
We can use the `t.Run` and group both the success and error test cases under the `TestForEachContextError` ([example](https://github.com/Shopify/hoff/blob/6084ba63d8e84f98b0b70e081f7c0328fc0a33af/map_test.go#L31-L54)).